pallets/author-mapping/src/lib.rs

```diff
@@ -195,16 +207,89 @@ pub mod pallet {
 
 			T::DepositCurrency::unreserve(&account_id, stored_info.deposit);
 
-			<Pallet<T>>::deposit_event(Event::AuthorDeRegistered { author_id });
+			<Pallet<T>>::deposit_event(Event::AuthorDeRegistered {
+				author_id,
+				account_id,
+				keys: stored_info.keys,
+			});
 
 			Ok(().into())
 		}
+
+		#[pallet::weight(<T as Config>::WeightInfo::add_association())] // TODO update weight
+		pub fn add_full_association(
+			origin: OriginFor<T>,
+			author_id: NimbusId,
+			keys: T::Keys,
+		) -> DispatchResult {
+			let account_id = ensure_signed(origin)?;
+
+			ensure!(
+				MappingWithDeposit::<T>::get(&author_id).is_none(),
+				Error::<T>::AlreadyAssociated
+			);
+
+			Self::enact_registration(&author_id, &account_id, keys.clone())?;
+
+			<Pallet<T>>::deposit_event(Event::AuthorRegistered {
+				author_id,
+				account_id,
+				keys,
+			});
+
+			Ok(())
+		}
+
+		/// Change your keys
+		///
+		/// This is useful for key rotation to update Nimbus and VRF keys in one call.
+		/// No new security deposit is required. Will replace `update_association` which is kept
+		/// now for backwards compatibility reasons.
+		#[pallet::weight(0)] // TODO: update weights
+		pub fn set_keys(
```

Comment:
This function name is missleanding. It looks as it is updating the association plus setting new keys. Can we replace if for something more like update_association_and_set_keys?

Reply:
`set_keys` sets all keys at once, which is the Author key and the VRF keys

`update_association` sets the new keys to the `new_author_id.into()` by default so it does set the keys as well